examples/openstack-notifications/queue_dump.py

```diff
@@ -73,32 +78,39 @@ def callback(ch, method, properties, body):
     if message:
         parsed["oslo.message"] = message
 
-    if args.event_type or args.provision_state:
-        if args.event_type:
-            event_type = message.get("event_type")
-            if event_type == args.event_type:
-                print(json.dumps(parsed, indent=4))
-
-        if args.provision_state:
-            payload = message.get("payload", {})
-            ironic_object = payload.get("ironic_object.data", {})
-            provision_state = ironic_object.get("provision_state")
-            if provision_state == args.provision_state:
-                print(json.dumps(parsed, indent=4))
-    else:
-        print(json.dumps(parsed, indent=4))
+    event_type = message.get("event_type")
 
-    if args.destroy:
-        print("Destroying message.")
-        # Acknowledge that the message has been handled, removing it from the queue.
-        ch.basic_ack(delivery_tag=method.delivery_tag)
+    if args.list_event_types:
+        if event_type:
+            print(event_type)
```

Comment:
```suggestion
    if args.list_event_types and message.get("event_type", None):
            print(event_type)
```

but also the help text suggests that this will print unique event types - where is this deduplication implemented?